src/main/java/picard/analysis/directed/RnaSeqMetricsCollector.java

```diff
@@ -430,20 +430,34 @@ else if (this.metrics.SAMPLE != null) {
             return out;
         }
 
+        protected Set<Gene> getGenesForPickTranscripts() {
+            return geneOverlapDetector.getAll();
+        }
+
         /** Picks the set of transcripts on which the coverage metrics are to be calculated. */
         public Map<Gene.Transcript, int[]> pickTranscripts(final Map<Gene.Transcript, int[]> transcriptCoverage) {
-            final Map<Gene.Transcript, Double> bestPerGene = new HashMap<Gene.Transcript, Double>();
+            // Get the set of genes for which there are transcripts
+            // Note that the size of transcriptCoverage may be << than size of genesForPickTranscripts, so iterating
+            // over that set should be much faster.  If the size is not much smaller, it shouldn't be significantly
+            // larger.
+            final Set<Gene> genesWithTranscripts = new HashSet<>();
```

Comment:
It might be clearer to move this into its own method, since the only value that's used after this block is `genesWithTranscripts`. Then the comment can be changed to a javadoc for the method.

Reply:
Good idea.